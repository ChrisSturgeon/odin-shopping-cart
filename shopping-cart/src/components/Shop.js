import '../styles/shop.css';
import { Link } from 'react-router-dom';
import Item from './Item';
import data from '../stock';

export default function Shop(props) {
  const totalVal = Math.round((props.totalPrice + Number.EPSILON) * 100) / 100;
  const catalogue = data;
  return (
    <div>
      <div>
        {props.totalPrice > 0 ? (
          <div className="summary">
            <div data-testid="totalItems">{props.items} items in cart</div>
            <div data-testid="totalValue">Â£{totalVal} total</div>
            <Link to="cart">
              <button>View Cart</button>
            </Link>
          </div>
        ) : (
          <div className="summary empty">
            <Link to="cart">
              <button>View Cart</button>
            </Link>
          </div>
        )}
      </div>
      <div className="cards">
        {catalogue.map((record) => {
          return (
            <Item
              key={record.sku}
              artwork={record.artwork}
              album={record.album}
              artist={record.artist}
              price={record.price}
              add={props.add}
              id={record.sku}
            />
          );
        })}
      </div>
    </div>
  );
}
