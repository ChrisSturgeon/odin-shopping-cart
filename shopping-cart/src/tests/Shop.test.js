import React from 'react';
import { render, screen } from '@testing-library/react';
import Shop from '../components/Shop';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('Shop tests', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Shop totalPrice={56} add={jest.fn()} />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders total value', () => {
    render(
      <BrowserRouter>
        <Shop totalPrice={56} add={jest.fn()} />
      </BrowserRouter>
    );
    const valueDiv = screen.getByTestId('totalValue');

    expect(valueDiv).toHaveTextContent('Â£56 total');
  });

  it('renders number of items', () => {
    render(
      <BrowserRouter>
        <Shop items={3} totalPrice={22} add={jest.fn()} />
      </BrowserRouter>
    );
    const valueDiv = screen.getByTestId('totalItems');

    expect(valueDiv).toHaveTextContent('3 items in cart');
  });
});
