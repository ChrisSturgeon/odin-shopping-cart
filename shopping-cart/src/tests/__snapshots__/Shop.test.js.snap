// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop tests matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="summary"
      >
        <div
          data-testid="totalItems"
        >
           items in cart
        </div>
        <div
          data-testid="totalValue"
        >
          £56 total
        </div>
        <a
          href="/cart"
        >
          <button>
            View Cart
          </button>
        </a>
      </div>
    </div>
    <div
      class="cards"
    >
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="a-love-supreme.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            A Love Supreme
          </div>
          <div
            class="artist"
          >
            John Coltrane
          </div>
          <div>
            £15.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="sound-of-silver.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Sound of Silver
          </div>
          <div
            class="artist"
          >
            LCD Soundsystem
          </div>
          <div>
            £19.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="illmatic.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Illmatic
          </div>
          <div
            class="artist"
          >
            Nas
          </div>
          <div>
            £19.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="alopecia.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Alopecia
          </div>
          <div
            class="artist"
          >
            WHY?
          </div>
          <div>
            £16.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="grace.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Grace
          </div>
          <div
            class="artist"
          >
            Jeff Buckley
          </div>
          <div>
            £17.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="woman.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Woman
          </div>
          <div
            class="artist"
          >
            Rhye
          </div>
          <div>
            £21.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="is-this-it.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Is This It
          </div>
          <div
            class="artist"
          >
            The Strokes
          </div>
          <div>
            £19.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="overgrown.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Overgrown
          </div>
          <div
            class="artist"
          >
            James Blake
          </div>
          <div>
            £14.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="amnesiac.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Amnesiac
          </div>
          <div
            class="artist"
          >
            Radiohead
          </div>
          <div>
            £15.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="doolittle.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Doolittle
          </div>
          <div
            class="artist"
          >
            The Pixies
          </div>
          <div>
            £13.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="dookie.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Dookie
          </div>
          <div
            class="artist"
          >
            Green Day
          </div>
          <div>
            £16.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
      <div
        class="item-card"
      >
        <img
          alt="record artwork"
          src="kind-of-blue.jpeg"
        />
        <div
          class="card-bottom"
        >
          <div>
            Kind of Blue
          </div>
          <div
            class="artist"
          >
            Miles Davis
          </div>
          <div>
            £12.99
          </div>
          <div
            class="card-qty"
          >
            <button>
              -
            </button>
            <div
              data-testid="quantity"
            >
              1
            </div>
            <button>
              +
            </button>
          </div>
          <button>
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
